deployment/src/main/java/io/quarkiverse/jasperreports/deployment/JasperReportsProcessor.java

```diff
@@ -413,12 +419,69 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
             } catch (IOException e) {
                 Log.error("Error looking for report files.", e);
             }
+        }
 
-            // Print the found files
-            foundFiles.forEach((file) -> {
-                reportFiles.produce(new ReportFileBuildItem(file));
-                watchedPaths.produce(new HotDeploymentWatchedFileBuildItem(file.toString()));
-            });
+        Log.debugf("Collected %s report file(s)", count.get());
+    }
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    void watchReportFiles(BuildProducer<HotDeploymentWatchedFileBuildItem> watchedPaths,
+            List<ReportFileBuildItem> reportFiles) {
+
+        // Print the found files
+        reportFiles.forEach((file) -> {
+            Log.tracef("Watching report file %s", file);
+            watchedPaths.produce(new HotDeploymentWatchedFileBuildItem(file.getFileName()));
+        });
+    }
+
+    @BuildStep
+    void compileReports(ReportConfig config, List<ReportFileBuildItem> reportFiles,
+            BuildProducer<GeneratedClassBuildItem> compiledReportProducer, OutputTargetBuildItem outputTarget) {
+
+        if (config.build().enable()) {
+            Log.debugf("Found %s report(s) to compile?", reportFiles.size());
+            try {
+                // make sure the destination path exists before we try to write files to it
+                Path outputFilePath = Path.of(outputTarget.getOutputDirectory().toString(), "classes",
+                        config.build().destination().get().toString());
+
+                if (!Files.exists(outputFilePath)) {
+                    Files.createDirectories(outputFilePath);
```

Comment:
You need to call `outputFilePath.getParent()` if I am not mistaken

Reply:
I did try that at first, but it didn't work right.  This one puts the output directory where I wanted it to be.